src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -312,17 +313,15 @@ void java_string_library_preprocesst::add_string_type(
     array_typet(java_char_type(), infinity_exprt(string_length_type())));
   string_type.add_base(symbol_typet("java::java.lang.Object"));
 
-  symbolt string_symbol;
-  string_symbol.name="java::"+id2string(class_name);
-  string_symbol.base_name=id2string(class_name);
-  string_symbol.pretty_name=id2string(class_name);
-  string_symbol.type=string_type;
-  string_symbol.is_type=true;
-
-  // Overwrite any pre-existing symbol in the table, e.g. created by
-  // a loaded model.
-  symbol_table.remove(string_symbol.name);
-  assert(!symbol_table.add(string_symbol));
+  std::string name="java::"+id2string(class_name);
+  symbolt tmp_string_symbol;
+  tmp_string_symbol.name=name;
```

Comment:
`name` is not used after this line - consider `tmp_string_symbol.name="java::"+id2string(class_name);`